src/components/SearchableSingleSelect/SearchableSingleSelect.tsx

```diff
@@ -0,0 +1,118 @@
+import { CircularLoader, Input, SingleSelect, SingleSelectOption } from '@dhis2/ui'
+import * as React from 'react'
+import { useCallback, useEffect, useState } from 'react'
+
+interface Option {
+    value: string,
+    label: string,
+}
+
+type OnFilterChange = ({ value }: { value: string }) => void
+type OnIntersectionChange = ({ isIntersecting }: { isIntersecting: boolean }) => void
+interface SearchableSingleSelectPropTypes {
+    onChange: ({ selected }: { selected: string }) => void,
+    onFilterChange: OnFilterChange,
+    onIntersectionChange: OnIntersectionChange,
+    options: Option[],
+    preventIntersectionDetection: boolean,
+    selected?: string,
+    showEndLoader: boolean,
+}
+
+export const SearchableSingleSelect = ({
+    onChange,
+    onFilterChange,
+    onIntersectionChange,
+    options,
+    preventIntersectionDetection,
+    selected,
+    showEndLoader,
```

Comment:
I think it would be ergonomic to have some `initialLoading`-prop here.
A common use-case in the list would be to have the "selected" but not the options yet. 
Since the selected option could be at any page, we would have to fetch that and the options in paralell (we would have the ID - eg from the url in header-bar). So would be great for a prop like that, that would show some loading-state and disable the select.